session_test.go

```diff
@@ -117,6 +118,95 @@ func TestRequestPriority(t *testing.T) {
 	}
 }
 
+func TestParseDirectedReadOption(t *testing.T) {
+	for _, tt := range []struct {
+		desc   string
+		option string
+		want   *pb.DirectedReadOptions
+	}{
+		{
+			desc:   "use directed read location option only",
+			option: "us-central1",
+			want: &pb.DirectedReadOptions{
+				Replicas: &pb.DirectedReadOptions_IncludeReplicas_{
+					IncludeReplicas: &pb.DirectedReadOptions_IncludeReplicas{
+						ReplicaSelections: []*pb.DirectedReadOptions_ReplicaSelection{
+							{
+								Location: "us-central1",
+							},
+						},
+						AutoFailoverDisabled: false,
+					},
+				},
+			},
+		},
+		{
+			desc:   "use directed read location and type option (RO)",
+			option: "us-central1:ro",
+			want: &pb.DirectedReadOptions{
+				Replicas: &pb.DirectedReadOptions_IncludeReplicas_{
+					IncludeReplicas: &pb.DirectedReadOptions_IncludeReplicas{
+						ReplicaSelections: []*pb.DirectedReadOptions_ReplicaSelection{
+							{
+								Location: "us-central1",
+								Type:     pb.DirectedReadOptions_ReplicaSelection_READ_ONLY,
+							},
+						},
+						AutoFailoverDisabled: false,
+					},
+				},
+			},
+		},
+		{
+			desc:   "use directed read location and type option (READ_ONLY)",
+			option: "us-central1:READ_ONLY",
+			want: &pb.DirectedReadOptions{
+				Replicas: &pb.DirectedReadOptions_IncludeReplicas_{
+					IncludeReplicas: &pb.DirectedReadOptions_IncludeReplicas{
+						ReplicaSelections: []*pb.DirectedReadOptions_ReplicaSelection{
+							{
+								Location: "us-central1",
+								Type:     pb.DirectedReadOptions_ReplicaSelection_READ_ONLY,
+							},
+						},
+						AutoFailoverDisabled: false,
+					},
+				},
+			},
+		},
+		{
+			desc:   "use directed read location and type option (RW)",
+			option: "us-central1:rw",
+			want: &pb.DirectedReadOptions{
+				Replicas: &pb.DirectedReadOptions_IncludeReplicas_{
+					IncludeReplicas: &pb.DirectedReadOptions_IncludeReplicas{
+						ReplicaSelections: []*pb.DirectedReadOptions_ReplicaSelection{
+							{
+								Location: "us-central1",
+								Type:     pb.DirectedReadOptions_ReplicaSelection_READ_WRITE,
+							},
+						},
+						AutoFailoverDisabled: false,
+					},
+				},
+			},
+		},
+		{
+			desc:   "use invalid type option",
+			option: "us-central1:rr",
+			want:   nil,
+		},
+	} {
+		t.Run(tt.desc, func(t *testing.T) {
+			got, _ := parseDirectedReadOption(tt.option)
+			//fmt.Println(got)
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
`cmp` package is safer alternative to `reflect`: https://pkg.go.dev/github.com/google/go-cmp/cmp

See an example: https://github.com/cloudspannerecosystem/spanner-cli/blob/4ad906e8b51f310456882a40ed0e25cdbbb119d5/cli_test.go#L109

Reply:
Thank you!
Fix with `!cmp.Equal(got, tt.want, protocmp.Transform())`